scan.go

```diff
@@ -155,12 +175,21 @@ func (c *ScanCmd) Execute(args []string) error {
 	return nil
 }
 
-func scan(pkgPatterns []string) ([]*unit.SourceUnit, error) {
+func isInGopath(path string) bool {
+	for _, gopath := range filepath.SplitList(buildContext.GOPATH) {
+		if strings.HasPrefix(evalSymlinks(path), filepath.Join(evalSymlinks(gopath), "src")) {
+			return true
+		}
+	}
+	return false
+}
+
+func scan(pkgPatterns []string, scanDir string) ([]*unit.SourceUnit, error) {
 	// TODO(sqs): include xtest, but we'll have to make them have a distinctly
 	// namespaced def path from the non-xtest pkg.
 
-	cmd := exec.Command(goBinaryName, "list", "-e", "-json")
-	cmd.Args = append(cmd.Args, pkgPatterns...)
+	// Go always evaluates symlinks in the working directory path, using sh is a workaround
+	cmd := exec.Command("sh", "-c", fmt.Sprintf(`cd %s; %s list -e -json %s`, scanDir, goBinaryName, strings.Join(pkgPatterns, " ")))
```

Comment:
No, that's exactly what i meant with the comment above the line. When you use `cmd.Dir`, then Go automatically evaluates all symlinks and the path `{builddir}/.srclib-gopath/src/{repo-url}` just becomes `{builddir}`. Should I improve the comment?